### PR TITLE
feat(logger): improve the logging via tracing

### LINKED ISSUE #645: Improve logging and store logs in a file
### Discussed in https://github.com/raphamorim/rio/discussions/626

<div type='discussions-op-text'>

<sup>Originally posted by **andrewcrook** August 31, 2024</sup>
I have tried to setup logging 

```
 [developer]
#log-level= “ERROR"
 log-level = “DEBUG"
```

However, I have looked in the common folders and console app for the log file and I cannot find its location.
Any ideas?</div>

I will be working on it so feel free to assign :)

### PR DESCRIPTION
**Pull request by orhun**
closes #645

A couple of points:

- Should we add a prefix path to the `log-file` option? For example, `log-file: rio.log` will be used as `~/.config/rio/rio.log`.
- Should the logs in the file also be filtered by the log level?
- Should we add a `--log-file` option to the CLI?

### CODE REVIEW

**Comment by raphamorim** [comment]
> Should we add a prefix path to the log-file option? For example, log-file: rio.log will be used as ~/.config/rio/rio.log.

Yes

> Should the logs in the file also be filtered by the log level?

Also yes

> Should we add a --log-file option to the CLI?

Optional but would be awesome to have

File: .gitignore [hunk 1]
```diff
@@ -14,3 +14,4 @@ NOTES
 *.wasm
 *.ts
 *.js
+rio.log
```
Comment (raphamorim) [inline .gitignore:17]: I think you can actually filter any log `*.log`

File: docs/docs/config/log-level.md [hunk 2]
```diff
@@ -3,11 +3,12 @@ title: 'log-level'
 language: 'en'
 ---
 
-This property enables log level filter. Default is "OFF".
+This property enables log level filter and file. The default level is "OFF" and the logs are not logged to a file as default.
 
 ```toml
 [developer]
 log-level = "OFF"
+log-file = "rio.log"
```
Comment (raphamorim) [inline docs/docs/config/log-level.md:11]: I think it should a boolean tbh, the name of the file should not change unless used another configuration property. What do you think?

**Comment by raphamorim** [comment]
Overall looks great, left few comments 👌

File: .gitignore (hunk 1 above)
Comment (orhun) [reply .gitignore]: 4150cb34a6f36a6a09636a6c66a6109f496fc6ba

File: docs/docs/config/log-level.md (hunk 2 above)
Comment (orhun) [reply docs/docs/config/log-level.md]: Done
cd4c280add17913b34ed78e42cafb9fa88943677

**Comment by orhun** [comment]
Ready for review again :)

**Review by raphamorim — APPROVED** [review]
Great addition to rio

**Pull request merged**